### PR TITLE
fix protocol documentation

### PR DESCRIPTION
**Pull request by MrMoon**
# Protocol Documentation Fix&Clean

### Motivation:
I was reading the documentation to learn more about rsocket, and I found some issues understanding the language, as well as some grammar and spelling issues. So I wanted to help fix them. Hope these are a good fix, please let me know if you want to make more clear, or even to do more of these.

### Modifications:
Fixed some sentences here and there.

### Result:
A clearer documentation

### CODE REVIEW

File: Protocol.md [hunk 1]
```diff
@@ -1,21 +1,24 @@
 ## Status
 
-This protocol is currently a draft for the final specifications. 
-Current version of the protocol is __0.2__ (Major Version: 0, Minor Version: 2). This is currently considered a 1.0 Release Candidate. Final testing is being done in Java and C++ implementations with goal to release 1.0 in the near future. 
+This protocol is currently a __draft__ for the final specifications. 
+Current version of the protocol is __0.2__ (Major Version: __0__, Minor Version: __2__). This is currently considered a 1.0 Release Candidate. 
+
+Final testing is being done in Java and C++ implementations with goal to release 1.0 in the near future.
```
Comment (rstoyanchev) [inline Protocol.md:6]: Let's remove this dated line altogether.

File: Protocol.md [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+This protocol is currently a __draft__ for the final specifications. 
+Current version of the protocol is __0.2__ (Major Version: __0__, Minor Version: __2__). This is currently considered a 1.0 Release Candidate. 
+
+Final testing is being done in Java and C++ implementations with goal to release 1.0 in the near future. 
 
 ## Introduction
 
-Specify an application protocol for [Reactive Streams](http://www.reactive-streams.org/) semantics across an asynchronous, binary
-boundary. For more information, please see [rsocket.io](http://rsocket.io/).
+RSocket, as the name implies, is a socket with the [Reactive Streams](http://www.reactive-streams.org/) semantics, that is where the 'R' comes from, across an asynchronous, binary boundary.
```
Comment (rstoyanchev) [inline Protocol.md:10]:
```suggestion
RSocket is an application protocol that provides [Reactive Streams](http://www.reactive-streams.org/) semantics across a network, asynchronous, binary boundary. 
```
I don't think we should lose the term "application protocol".

File: Protocol.md [hunk 3]
```diff
@@ -74,8 +77,7 @@ It is up to a server to accept clients of lower versions than what it supports.
 
 ## Data And Metadata
 
-RSocket provides mechanisms for applications to distinguish payload into two types. Data and Metadata. The distinction
-between the types in an application is left to the application. 
+RSocket provides mechanisms for applications to distinguish a payload into two types. Data and Metadata, this distinction is left to the application.
```
Comment (rstoyanchev) [inline Protocol.md:80]:
```suggestion
RSocket provides mechanisms for applications to differentiate between two types of content within a payload, Data and Metadata, with the distinction left to applications to make use of according to their needs.
```

File: Protocol.md [hunk 4]
```diff
@@ -90,9 +92,9 @@ The following are features of Data and Metadata.
 
 The RSocket protocol uses a lower level transport protocol to carry RSocket frames. A transport protocol MUST provide the following:
 
-1. Unicast [Reliable Delivery](https://en.wikipedia.org/wiki/Reliability_(computer_networking)).
-1. [Connection-Oriented](https://en.wikipedia.org/wiki/Connection-oriented_communication) and preservation of frame ordering. Frame A sent before Frame B MUST arrive in source order. i.e. if Frame A is sent by the same source as Frame B, then Frame A will always arrive before Frame B. No assumptions about ordering across sources is assumed.
-1. [FCS](https://en.wikipedia.org/wiki/Frame_check_sequence) is assumed to be in use either at the transport protocol or at each MAC layer hop. But no protection against malicious corruption is assumed.
+1. Unicast [Reliable Delivery](https://en.wikipedia.org/wiki/Reliability_(computer_networking)) Delivery. Meaning that it's a one-to-one transmission from one point in the network to another point; that is, one sender and one receiver. In addition to being "at least once", i.e. at least one copy of the message is guaranteed to be delivered to the recipient.
```
Comment (rstoyanchev) [inline Protocol.md:95]:
```suggestion
1. Unicast [Reliable Delivery](https://en.wikipedia.org/wiki/Reliability_(computer_networking)), i.e. between a single sender and a single receiver with "at least once" message delivery guarantees.
```
"Delivery" is duplicated (once i the link and once after the link).

File: Protocol.md [hunk 5]
```diff
@@ -90,9 +92,9 @@ The following are features of Data and Metadata.
...
+1. [Connection-Oriented](https://en.wikipedia.org/wiki/Connection-oriented_communication) along with preservation of frame ordering. if Frame A is sent by the same source as Frame B, then Frame A will always arrive before Frame B. No assumptions about ordering across sources is assumed.
```
Comment (rstoyanchev) [inline Protocol.md:96]: "if" should be capitalized (start of sentence).

**Review by rstoyanchev — CHANGES_REQUESTED** [review]

File: Protocol.md (hunk 1 above)
Comment (MrMoon) [reply Protocol.md:6]: Should I also remove the `Status` Heading ?

File: Protocol.md [hunk 6]
```diff
@@ -245,22 +239,22 @@ When the max Stream ID has been used:
...
+    - If all Stream IDs are concurrently in use, no new streams can be created, thus a new connection MUST be established to create new streams.
 
 It is RECOMMENDED that Stream ID re-use only be used in combination with resumability. 
 
 ### Frame Types
 
 |  Type                          | Value  | Description |
 |:-------------------------------|:-------|:------------|
-| __RESERVED__                                     | 0x00 | __Reserved__ |
+| __RESERVED__                                     | 0x00 | __Reserved__ for any operation involving the connection. |
```
Comment (yschimke) [inline Protocol.md:257]: This is not correct.  `Stream ID of value __0__` but this section is discussing the frame types, not stream ids.

Comment (MrMoon) [reply Protocol.md:257]: oh !!!! got it
```suggestion
| __RESERVED__                                     | 0x00 | __Reserved__ |
```

Comment (MrMoon) [reply Protocol.md:257]: oh !!! got it 👍🏽

**Comment by yschimke** [comment]
LGTM - as I'm less actively involved in the core of rsocket these days, I'll leave approval to others.

**Pull request closed without merging**